WordPress/src/main/java/org/wordpress/android/viewmodel/PluginBrowserViewModel.kt

```diff
@@ -0,0 +1,465 @@
+package org.wordpress.android.viewmodel
+
+import android.arch.lifecycle.LiveData
+import android.arch.lifecycle.MutableLiveData
+import android.arch.lifecycle.ViewModel
+import android.os.Bundle
+import android.os.Handler
+import android.support.annotation.WorkerThread
+import android.text.TextUtils
+import org.greenrobot.eventbus.Subscribe
+import org.greenrobot.eventbus.ThreadMode
+import org.wordpress.android.WordPress
+import org.wordpress.android.fluxc.Dispatcher
+import org.wordpress.android.fluxc.generated.PluginActionBuilder
+import org.wordpress.android.fluxc.model.SiteModel
+import org.wordpress.android.fluxc.model.plugin.ImmutablePluginModel
+import org.wordpress.android.fluxc.model.plugin.PluginDirectoryType
+import org.wordpress.android.fluxc.store.PluginStore
+import org.wordpress.android.fluxc.store.PluginStore.FetchPluginDirectoryPayload
+import org.wordpress.android.util.AppLog
+import org.wordpress.android.util.AppLog.T
+import org.wordpress.android.util.StringUtils
+import java.util.ArrayList
+import java.util.HashMap
+import java.util.HashSet
+import javax.inject.Inject
+import kotlin.properties.Delegates
+
+private const val KEY_SEARCH_QUERY = "KEY_SEARCH_QUERY"
+private const val KEY_TITLE = "KEY_TITLE"
+
+@WorkerThread
+class PluginBrowserViewModel @Inject
+constructor(private val mDispatcher: Dispatcher, private val mPluginStore: PluginStore) : ViewModel() {
+    enum class PluginListType {
+        SITE,
+        FEATURED,
+        POPULAR,
+        NEW,
+        SEARCH
+    }
+
+    enum class PluginListStatus {
+        CAN_LOAD_MORE,
+        DONE,
+        ERROR,
+        FETCHING,
+        LOADING_MORE
+    }
+
+    private var isStarted = false
+
+    private val handler = Handler()
+    private val updatedPluginSlugSet = HashSet<String>()
+
+    private val _featuredPlugins = MutableLiveData<List<ImmutablePluginModel>>()
```

Comment:
I really don't like that I needed a backing field for these `LiveData` properties. I didn't want to expose these properties as `Mutable` to the Activity/Fragment, so for now I thought it was the best way to go. However I am working on an idea that will get rid of this, as well as the statuses (actually combine them), so it'll look much better in the next PR. I decided not to dwell on it too much for now, but I'd love to learn if there is a better way to do this.

Reply:
AFAIK there is no way around it.